website/docs/r/identity_oidc_client.html.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: "vault"
+page_title: "Vault: vault_identity_oidc_client resource"
+sidebar_current: "docs-vault-identity-oidc-client"
+description: |-
+    Provision OIDC Clients in Vault.
+---
+
+# vault\_identity\_oidc\_client
+
+Manages OIDC Clients in a Vault server. See the [Vault documentation](https://www.vaultproject.io/api-docs/secret/identity/oidc-provider#create-or-update-an-assignment)
+for more information.
+
+## Example Usage
+
+```hcl
+resource "vault_identity_oidc_key" "test" {
+  name               = "default"
+  allowed_client_ids = ["*"]
+  rotation_period    = 3600
+  verification_ttl   = 3600
+}
+
+resource "vault_identity_oidc_assignment" "test" {
+  name       = "my-assignment"
+  entity_ids = ["ascbascas-2231a-sdfaa"]
+  group_ids  = ["sajkdsad-32414-sfsada"]
+}
+
+
+resource "vault_identity_oidc_client" "test" {
+  name          = "my-app"
+  key           = vault_identity_oidc_key.default.name
+  redirect_uris = [
+    "http://127.0.0.1:9200/v1/auth-methods/oidc:authenticate:callback",
+    "http://127.0.0.1:8251/callback",
+    "http://127.0.0.1:8080/callback"
+  ]
+  assignments = [
+    vault_identity_oidc_assignment.test.name
+  ]
+  id_token_ttl     = 2400
+  access_token_ttl = 7200
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the client.
+
+* `key` - (Required) A reference to a named key resource in Vault.
+  This cannot be modified after creation.
+
+* `redirect_uris` - (Optional) Redirection URI values used by the client. 
+  One of these values must exactly match the `redirect_uri` parameter value
+  used in each authentication request.
+
+* `assignments` - (Optional) A list of assignment resources associated with the client.
+
+* `id_token_ttl` - (Optional) The time-to-live for ID tokens obtained by the client. 
+  The value should be less than the `verification_ttl` on the key.
+
+* `access_token_ttl` - (Optional) The time-to-live for access tokens obtained by the client.
```

Comment:
Similar to comment above, `client_type` can be added here as a field.

Reply:
Added docs for new field in 1e9f8a3